README_DEVELOPERS.md

```diff
@@ -1,17 +1,87 @@
 
-## Install dev tools
+It is split into two parts, the backend logic in `cd-service`, and the frontend which is split into two microservices, the `frontend-service` which provides the rest backing for the ui, and the `ui-service` with the actual ui. 
+The `cd-service` takes the url of the repository to watch from the environment variable `KUBERPULT_GIT_URL` and the branch to watch from the environment variable `KUBERPULT_GIT_BRANCH`.
 
-- libgit2 >= 1.0
+## pre requisite software
+
+- [docker](https://docs.docker.com/get-docker/)
+- [docker-compose](https://docs.docker.com/compose/install/)
+
+## setup and run instructions (with docker compose)
+
+- in `services/cd-service`, initialize a bare repository with the name `repository_remote`
+
+```bash
+cd services/cd-service
+git init --bare repository_remote
+cd ../..
+```
+- the value of environment variables are defaulted to `KUBERPULT_GIT_URL=./repository_remote` and `KUBERPULT_GIT_BRANCH=master`
+- run the following command to start all the services required, the `--build` parameter is added to build any changes you may have added to the code
+
+```bash
+docker compose up --build
+```
+- the `cd-service` is available at `localhost:8080` and the kuberpult ui is available at `localhost:3000`
+
+
+### to test setup was done correctly
+
+- for adding changes and testing releasing, clone the `repository_remote` folder. 
+- calling curl command to `/release` api with form data for manifest file should have update the remote repository with a new relase.
+- view the changes in ui as well
+
+```bash
+cd services/cd-service
+git clone ./repository_remote repository_checkedout
+cd repository_checkedout
+touch manifest.yaml
+# This should cause the release to be pushed to the git repository
+curl --form-string 'application=helloworld' --form 'manifests[development]=@manifest.yaml' localhost:8080/release
+git pull
+cd ../../..
+```
+
+## Unit tests
+
+Go tests would be part of the same package as the main code, but ending the file names with `_test.go`. When adding new testcases, please use [table driven tests](https://revolution.dev/app/-JqFGExX46gs9mH7vxR5/WORKSPACE_DOCUMENT/-MjkBXy5_eugWYQsxyHl/) 
+
+To run tests, the root makefile has test command which runs the test commands in `services/cd-service/Makefile` and `services/frontend-service/Makefile`, which in turn run tests for go and yarn files.
+
+```bash
+make test
+```
+
+When there is build issues in the test code, it would show up as a build failure during make test with the proper error.
+
+When a single test case fails, the test case shows up with the curresponding error.
+
+For a more verbose version, you could go into the service directory and run the tests manually in verbose mode
 
-  Download tar file and follow instructions here: https://github.com/libgit2/libgit2#installation
-  It worked for me to run: (the instructions are slightly different)
+```bash
+cd services/cd-service
+go test ./... -v
+```
+
+# Legacy docs 
```

Comment:
How is this legacy? We don't intend to get rid of yarn or node. I would call this "Installation outside of docker"